### PR TITLE
Fix races in build image trigger int test

### PR DESCRIPTION
**Pull request by ironcladlou**
Eliminate the following races from the test:

* Establish a watch prior to API Create operations, and consume the initial
  event resulting from Create before proceeding
* Consume watch events for updates rather than relying on a GET

### CODE REVIEW

**Comment by ironcladlou** [comment]
[test]

File: test/integration/imagechange_buildtrigger_test.go [hunk 1]
```diff
@@ -42,17 +42,17 @@ func TestSimpleImageChangeBuildTrigger(t *testing.T) {
...
-	}
-
-	watch2, err := openshift.Client.BuildConfigs(testNamespace).Watch(labels.Everything(), labels.Everything(), created.ResourceVersion)
+	watch2, err := openshift.Client.BuildConfigs(testNamespace).Watch(labels.Everything(), labels.Everything(), "0")
 	if err != nil {
 		t.Fatalf("Couldn't subscribe to BuildConfigs %v", err)
 	}
 	defer watch2.Stop()
 
+	if _, err := openshift.Client.BuildConfigs(testNamespace).Create(config); err != nil {
```
Comment (smarterclayton) [inline test/integration/imagechange_buildtrigger_test.go]: Why did you move this?  Watch from resource version shouldn't matter order.

Comment (smarterclayton) [inline test/integration/imagechange_buildtrigger_test.go]: Also, this is less resilient to other problems (because of watch from 0).  I thought ben explained that there really were two changes, so we should test those.

**Comment by smarterclayton** [comment]
```
Previous write by goroutine 628:
  sync.raceWrite()
      /home/travis/.gvm/gos/go1.3.3/src/pkg/sync/race.go:41 +0x35
  sync.(*WaitGroup).Wait()
      /home/travis/.gvm/gos/go1.3.3/src/pkg/sync/waitgroup.go:122 +0x176
  net/http/httptest.(*Server).Close()
      /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:168 +0x7e
  github.com/openshift/origin/test/integration.(*testOpenshift).Close()
      /home/travis/gopath/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/test/integration/deploy_trigger_test.go:344 +0x95
  github.com/openshift/origin/test/integration.TestSimpleImageChangeBuildTrigger()
      /home/travis/gopath/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/test/integration/imagechange_buildtrigger_test.go:81 +0x11ee
  testing.tRunner()
      /home/travis/.gvm/gos/go1.3.3/src/pkg/testing/testing.go:422 +0x10f
```

Is this what you're fixing?

File: test/integration/imagechange_buildtrigger_test.go (hunk 1 above)
Comment (ironcladlou) [inline test/integration/imagechange_buildtrigger_test.go]: I was getting sporadic Added events on the watch channel with the previous ordering. Didn't dig deeper. That explains the dual-watches. I don't see any explanation as to why there should be an Added AND a Modified event on that channel when the watch was established AFTER the Create and from the created resource's version. Something is going on here.

**Comment by ironcladlou** [comment]
Haven't seen the "previous write" error. I was getting other failures due to the sporadic/unwanted Added BuildConfig event.

**Comment by smarterclayton** [comment]
Why is it sporadic?  I don't think we've answered that - ben seems to think there *should* be two writes.  I want the test to be more deterministic (not less)
